rs/recovery/src/nns_recovery_failover_nodes.rs

```diff
@@ -76,9 +80,11 @@ pub struct NNSRecoveryFailoverNodesArgs {
     #[clap(long)]
     pub download_node: Option<IpAddr>,
 
-    /// IP address of the node to upload the new subnet state to
-    #[clap(long)]
-    pub upload_node: Option<IpAddr>,
+    /// The method of uploading state. Possible values are either `local` (for a
+    /// local recovery on the admin node) or the ipv6 address of the target node.
```

Comment:
```suggestion
    /// local recovery on an admin node) or the ipv6 address of the target node.
```

Reply:
> As far as I can tell the tests are only using remote recoveries now. Is there a way to test the new local functionality as well?

Yeah, I'm not sure how we would be able to achieve that. I don't think the test driver can run on the node. Perhaps ssh-ing into the node and manually driving the ic-recovery tool, but that sounds very hacky. Any thoughts? @eichhorl

Reply:
Yes, the reality is that we didn't test the local recoveries before either, which is likely the reason for the minor road bumps we tend to run into whenever we do it live. 

We have [CON-1407](https://dfinity.atlassian.net/browse/CON-1407) to change this, but ssh-ing into the node and manually running `ic-recovery` is also the only way that came to my mind so far.

The good news is that technically, you should be able to specify all the arguments that we usually enter "interactively" already beforehand, as CLI arguments to the `ic-recovery` binary (similar to how it is done for the [mainnet NNS testnet](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/tests/nns/ic_mainnet_nns_recovery/src/lib.rs?L807-845)). So hopefully, instead of "driving" the tool start to finish, it would be enough to figure out all the arguments in the test driver beforehand, and then simply send one large recovery command to the admin node via SSH.

Reply:
Makes sense. If we could perform a recovery non-interactively with a single command, that would be a big win. Though we'd need an option to auto-select the download node, since we can't know beforehand which node will have the highest metrics.

Reply:
I think we could still break the subnet and pick the download node before we start the recovery. In a way that's what we do in the existing system tests as well: https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/tests/consensus/subnet_recovery/common.rs?L396-413